lib/rust/ensogl/core/src/system/gpu/shader/compiler.rs

```diff
@@ -82,11 +82,12 @@ const MAX_PARALLEL_COMPILE_JOBS: usize = 2;
 #[derivative(Debug)]
 pub struct Job<T> {
     #[derivative(Debug = "ignore")]
-    on_ready: Box<dyn FnOnce(shader::Program)>,
+    on_ready:  Box<dyn FnOnce(shader::Program)>,
     #[deref]
-    input:    T,
-    handler:  WeakJobHandler,
-    profiler: profiler::Debug,
+    input:     T,
+    handler:   WeakJobHandler,
+    cache_key: ShaderCacheKey,
```

Comment:
the name `cache_key` is not explanatory. It needs either better name or documentation.